lms/djangoapps/instructor/views/tools.py

```diff
@@ -61,14 +62,13 @@ def get_student_from_identifier(unique_student_identifier):
 
     Returns the student object associated with `unique_student_identifier`
 
-    Raises User.DoesNotExist if no user object can be found.
+    Raises User.DoesNotExist if no user object can be found, the user was
+    retired, or the user is in the process of being retired.
+
+    DEPRECATED: use student.models.get_user_by_username_or_email instead.
     """
-    unique_student_identifier = strip_if_string(unique_student_identifier)
-    if "@" in unique_student_identifier:
-        student = User.objects.get(email=unique_student_identifier)
-    else:
-        student = User.objects.get(username=unique_student_identifier)
-    return student
+    # TODO: add deprecation warning
```

Comment:
Can be removed?

Reply:
if you're referring to removing this entire function, it's referenced in like 20 other places.  I didn't want to flood this PR with those updates, but at least leave things better than the way they were before.  Besides, the underlying implementation is flawed, so this would need to be re-worked again anyway.